spark/src/main/scala/io/delta/tables/execution/VacuumTableCommand.scala

```diff
@@ -32,28 +33,15 @@ import org.apache.spark.sql.types.StringType
  * }}}
  */
 case class VacuumTableCommand(
-    path: Option[String],
-    table: Option[TableIdentifier],
+    override val child: LogicalPlan,
     horizonHours: Option[Double],
-    dryRun: Boolean) extends LeafRunnableCommand {
+    dryRun: Boolean) extends RunnableCommand with UnaryLike[LogicalPlan]{
 
   override val output: Seq[Attribute] =
     Seq(AttributeReference("path", StringType, nullable = true)())
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
-    val pathToVacuum =
-      if (path.nonEmpty) {
-        new Path(path.get)
-      } else if (table.nonEmpty) {
-        DeltaTableIdentifier(sparkSession, table.get) match {
-          case Some(id) if id.path.nonEmpty =>
-            new Path(id.path.get)
-          case _ =>
-            new Path(sparkSession.sessionState.catalog.getTableMetadata(table.get).location)
-        }
-      } else {
-        throw DeltaErrors.missingTableIdentifierException("VACUUM")
-      }
+    val pathToVacuum = getDeltaTable(child, "VACUUM").path
     val baseDeltaPath = DeltaTableUtils.findDeltaTableRoot(sparkSession, pathToVacuum)
```

Comment:
I don't think we need to find table root any more with this change? If the `child` pointed to a subdirectory of the table, I would have expected an AnalysisException before now. Query resolution would not have been able to turn `UnresolvedDeltaPathOrIdentifier` into a Delta table (because no `_delta_log` directory present in the expected location). 

If we _really_ need to support triggering VACUUM for a table by pointing at any subdirectory of that table (as the current code does), then we'd have to somehow delay the table resolution until this point so we can `findDeltaTableRoot`. But allowing users to specify subdirectories, as if they were the table itself, seems more like a bug than a feature, to be honest.

And actually, L60 below seems to corroborate that subdirectories aren't supported that way, because it blows up if the found root path mismatches the given table path?

Reply:
Thanks for the suggestion. I removed the baseDeltaPath check

Reply:
Hmm, we should probably double check the existing behavior first -- if vacuuming a subdirectory was supported before, and our changes here would block it, then that's a breaking change and we need to proceed very carefully.

CC @tdas @zsxwing